src/mono/wasm/runtime/driver.c

```diff
@@ -969,6 +969,28 @@ mono_wasm_get_type_aqn (MonoType * typePtr) {
 	return mono_type_get_name_full (typePtr, MONO_TYPE_NAME_FORMAT_ASSEMBLY_QUALIFIED);
 }
 
+EMSCRIPTEN_KEEPALIVE int
+mono_wasm_read_value_if_bool_unsafe (PVOLATILE(MonoObject) obj) {
```

Comment:
suggestions?

Reply:
I could also add more comments, the use-case is give me bool value if it's a bool, otherwise `null` on JS side. Then fallback to reading it as a string in Blazor.

Reply:
how about read_bool_or_null_unsafe?

Reply:
incidentally, from reading this if 'obj' is null it will return MARSHAL_ERROR_NULL_CLASS_POINTER, is that intended? Shouldn't it return -1 like the type check failure does?

Reply:
Maybe `mono_wasm_try_read_bool_value_unsafe` - to indicate that it will try to return a bool (0/1) value, but there is a failure case in which it returns `-1`.

Reply:
> how about read_bool_or_null_unsafe?

oops, I posted before I saw this.